core/modules/node/node.pages.inc

```diff
@@ -628,3 +628,90 @@ function node_revision_delete_confirm_submit($form, &$form_state) {
     $form_state['redirect'] .= '/revisions';
   }
 }
+
+/**
+ * Autocomplete callback for nodes by title.
+ *
+ * Searches for a node by title, but then identifies it by nid, so the actual
+ * returned value can be used later by the form.
+ *
+ * The returned $matches array has
+ * - key: The title, with the identifying nid in brackets, like "Some node
+ *   title [3325]"
+ * - value: the title which will is displayed in the autocomplete pulldown.
+ *
+ * Note that we must use a key style that can be parsed successfully and
+ * unambiguously. For example, if we might have node titles that could have
+ * [3325] in them, then we'd have to use a more restrictive token.
+ *
+ * @param string $string
+ *   The string that will be searched.
+ */
+function node_autocomplete($string = "") {
+  $matches = array();
+  if ($string) {
+    $result = db_select('node')
+      ->fields('node', array('nid', 'title'))
+      ->condition('title', db_like($string) . '%', 'LIKE')
+      ->range(0, 10)
```

Comment:
We should add the node access addition here:
```
      ->addTag('node_access')
```

That will prevent a user from auto-completing nodes to which they do not have access.